core/src/subgraph/registrar.rs

```diff
@@ -416,61 +387,62 @@ where
         AssignmentEvent::Add {
             subgraph_id,
             node_id: _,
-        } => Box::new(start_subgraph(subgraph_id, &*provider, logger).map_err(|()| unreachable!())),
+        } => {
+            start_subgraph(subgraph_id, provider.clone(), logger).await;
+            Ok(())
```

Comment:
Is the extra `Ok(())` needed or could we just return the `.await` result?

Reply:
start_subgraph doesn't return Result, so an Ok is needed. I moved it up to just wrap the outer value though.